File: archived/src/go/ch01/1.3/l1/demo_number_theory.go
package main

import "fmt"

func main() {
	// GCD / LCM
	a, b := int64(252), int64(198)
	fmt.Printf("gcd(%d, %d) = %d\n", a, b, GCD(a, b))
	fmt.Printf("lcm(12, 18) = %d\n", LCM(12, 18))

	// Modular identities
	x, y, m := int64(123), int64(456), int64(7)
	fmt.Printf("(x+y)%%m vs ModAdd: %d %d\n", (x+y)%m, ModAdd(x, y, m))
	fmt.Printf("(x*y)%%m vs ModMul: %d %d\n", (x*y)%m, ModMul(x, y, m))

	// Fraction reduction
	r := ReduceFraction(84, 126)
	fmt.Printf("reduceFraction(84,126) = %d/%d\n", r.Num, r.Den)

	// Exercises (optional to show here)
	runExercises()
}

File: archived/v1/src/erlang/ch01/1.3/l1/src/demo_number_theory.erl
-module(demo_number_theory).
-export([main/0]).

main() ->
    %% GCD / LCM
    A = 252, B = 198,
    io:format("gcd(~p, ~p) = ~p~n", [A, B, gcd_lcm:gcd(A,B)]),
    io:format("lcm(12, 18) = ~p~n", [gcd_lcm:lcm(12, 18)]),

    %% Modular identities
    X = 123, Y = 456, M = 7,
    io:format("(x+y)%%m vs mod_add: ~p ~p~n", [(X+Y) rem M, modular_identities:mod_add(X,Y,M)]),
    io:format("(x*y)%%m vs mod_mul: ~p ~p~n", [(X*Y) rem M, modular_identities:mod_mul(X,Y,M)]),

    %% Fraction reduction
    {Num,Den} = fractions_utils:reduce_fraction({84,126}),
    io:format("reduce_fraction(84,126) = ~p/~p~n", [Num, Den]).